File: workflow.post.js
const { Octokit } = require("@octokit/rest");

var token = process.env.GH_TOKEN;

var octokit = new Octokit({ auth: token });

var myinput = {"p1": "one", "p2": "two"};

var myinputs = { 
 p1: "js",
}

//inputs: JSON.stringify(myinput)

octokit.request('POST /repos/{owner}/{repo}/actions/workflows/{workflow_id}/dispatches', {
    owner: 'munderseth',
    repo: 'my.dispatch',
    workflow_id: 'wf.yml',
    ref: 'master',
    inputs: myinputs,
  })

File: workflow.post.py
import sys
import requests
import os
import json

BRANCH="master"

url      = "https://api.github.com/repos/munderseth/my.dispatch/actions/workflows/wf.yml/dispatches"
GH_TOKEN  = os.getenv('GH_TOKEN')

headers = {
  'Content-Type': "application/json",
  'Accept': "application/vnd.github.v3+json",
  'Authorization': 'Token ' + GH_TOKEN,
}

data = {
  "ref": BRANCH,
  "inputs": {"p1": "py"} 
}

def main():
 
  payload = json.dumps(data)
  print(payload)
  
  response = requests.request("POST", url, data=payload, headers=headers)
  print(response.status_code)

if __name__ == "__main__":  # pragma: no cover
    main()
